core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -114,28 +123,45 @@ class HistoryServer(
    * this UI with the event logs in the provided base directory.
    */
   def initialize() {
-    attachPage(new HistoryPage(this))
+    if (!initialized.getAndSet(true)) {
```

Comment:
It's in `HistoryServerSuite`

```scala
server = new HistoryServer(conf, provider, securityManager, 18080)
server.initialize()
server.bind()
```

issue here is the HS constructor already calls `initialize()`; some of the tests call it twice.

One option here would be move that `initialize()` out of the constructor of HS; that'd mean changing `main()`, but produce something better for subclassing and testing anyway. Even so, making `initialize()`  non-reentrant can only be a good thing, given that threads are spawned off, metrics registered, etc.